### PR TITLE
Context and lifecycle overhaul

### PR DESCRIPTION
**Pull request by nathan-knight**
I have updated this library to use the new React Context API and made it strict mode compliant by eliminating the use of UNSAFE_ lifecycle methods. I also improved the typings in places that I could.

This was done for use in a project I am working on but if the maintainer is interested in bringing this into the original please let me know of any required changes.

I tested it by going through the Storybook but given the scope of the changes there are probably regressions.

#### Checklist
- [x] My commits follow [Conventional Commits](https://www.conventionalcommits.org)
- [x] documentation is updated

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,7 +67,7 @@ npm install react-financial-charts
 
 Refer to [CONTRIBUTING.md](./CONTRIBUTING.md)
 
-This project is a mono-repo that uses [Lerna](https://lernajs.io/) to manage dependencies between packages.
+This project is a mono-repo that uses [Lerna](https://lerna.js.org/) to manage dependencies between packages.
```
Comment (nathan-knight) [inline README.md:70]: This was the wrong URL

**Comment by markmcdowell** [comment]
Thanks for the PR! I know that would have been a pain to do so happy to accept and we can test edge cases once merged.
